app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -349,6 +357,62 @@ class _LinkRoomPageConsumerState extends ConsumerState<LinkRoomPage> {
       final room = await ref.watch(maybeRoomProvider(roomId).future);
       if (room != null) {
         room.addParentRoom(selectedParentSpaceId, true);
+        final joinRule = room.joinRuleStr();
+        List<String> currentRooms = [];
+        bool parentCanSee = joinRule == 'public';
+        String newRule = 'restricted';
+        if (joinRule == 'restricted' || joinRule == 'knock_restricted') {
+          currentRooms =
+              room.restrictedRoomIdsStr().map((t) => t.toString()).toList();
+          parentCanSee = currentRooms.contains(selectedParentSpaceId);
+          newRule = joinRule;
+        }
+
+        if (!parentCanSee) {
+          // ignore: use_build_context_synchronously
+          bool shouldChange = await showDialog(
+            context: context,
+            builder: (context) {
+              return AlertDialog(
+                title: const Text('Not visible'),
+                content: const Wrap(
+                  children: [
+                    Text(
+                      "The current join rules of the child mean it won't be visible in the parent space to non-members. Should we update the join rules to allow for any parent space member to see and join the sub space?",
```

Comment:
Suggestion:
Although, this feature is mainly used by admin or pro users, we can make this make less technical and more user friendly.

Reply:
Only if you have a better suggestion...

Reply:
Just a suggestion. You can modify or use it based on what best fit for the requirement.

"The current join rules of the {roomName} mean it won't be visible in the {parentSpaceName}'s space members. Should we update the join rules to allow for {parentSpaceName}'s space member to see and join the {roomName}?"